ci/mkpipeline.py

```diff
@@ -390,6 +393,34 @@ def visit(step: PipelineStep) -> None:
     ]
 
 
+def trim_builds(pipeline: Any, coverage: bool) -> None:
+    """Trim unnecessary x86-64/aarch64 builds if all artifacts already exist."""
+
+    def builds_published(arch: Arch) -> bool:
+        repo = mzbuild.Repository(Path("."), arch=arch, coverage=coverage)
+        deps_publish = repo.resolve_dependencies(
+            image for image in repo if image.publish
+        )
+        return deps_publish.check()
+
+    def visit(step: dict[str, Any]) -> None:
+        if step.get("id") == "build-x86_64":
+            if builds_published(Arch.X86_64):
+                step["skip"] = True
+        if step.get("id") == "build-aarch64":
+            branch = os.environ["BUILDKITE_BRANCH"]
+            if (
```

Comment:
Why are the checks for the two architectures asymmetrical?

Reply:
We already don't run build-aarch64 for PRs, only on main and v*.* branches.